src/tiles/hover/hover-pane-controller.ts

```diff
@@ -0,0 +1,463 @@
+import {
+  html,
+  HTMLTemplateResult,
+  nothing,
+  ReactiveController,
+  ReactiveControllerHost,
+} from 'lit';
+import type { TileModel } from '../../models';
+
+type HoverPaneState = 'hidden' | 'shown' | 'fading-out';
+
+export interface HoverPaneProperties {
+  model?: TileModel;
+  baseNavigationUrl?: string;
+  baseImageUrl?: string;
+  loggedIn: boolean;
+  sortParam: SortParam | null;
+  collectionNameCache?: CollectionNameCacheInterface;
+}
+
+export interface HoverPaneControllerOptions {
+  offsetX?: number;
+  offsetY?: number;
+  enableLongPress?: boolean;
+  showDelay?: number;
+  hideDelay?: number;
+  longPressDelay?: number;
+  mobileBreakpoint?: number;
+}
+
+/** A common interface for providing a hover pane element. */
+export interface HoverPaneProviderInterface {
+  /** Returns the provider's currently rendered hover pane element. */
+  getHoverPane(): HTMLElement | undefined;
+  /** Returns properties that should be passed to the hover pane. */
+  getHoverPaneProps(): HoverPaneProperties;
+}
+
+/**
+ * An interface for interacting with hover pane controllers (e.g.,
+ * to retrieve their current hover pane template).
+ */
+export interface HoverPaneControllerInterface extends ReactiveController {
+  /**
+   * Returns the hover pane template to render based on this controller's
+   * current state. The returned template may be `nothing` if the hover
+   * pane should not currently be rendered.
+   */
+  getTemplate(): HTMLTemplateResult | typeof nothing;
+}
+
+const clamp = (val: number, min = -Infinity, max = Infinity) =>
+  Math.max(min, Math.min(val, max));
+
+export class HoverPaneController implements HoverPaneControllerInterface {
+  /**
+   * The hover pane element attached to this controller's host.
+   */
+  private hoverPane?: HTMLElement;
+
+  /**
+   * The properties to be passed to the hover pane element
+   */
+  private hoverPaneProps?: HoverPaneProperties;
+
+  /**
+   * The breakpoint (in pixels) below which the mobile interface should be used.
+   */
+  private mobileBreakpoint?: number = 600;
+
+  /**
+   * The number of horizontal pixels the hover pane should be offset from the
+   * pointer position.
+   */
+  private offsetX: number = -10;
+
+  /**
+   * The number of vertical pixels the hover pane should be offset from the
+   * pointer position.
+   */
+  private offsetY: number = 15;
+
+  /**
+   * The delay between the mouse idling within the host element and when the hover
+   * pane should begin fading in (in milliseconds).
+   */
+  private showDelay: number = 300;
+
+  /**
+   * The delay between when the mouse leaves the host element and when the hover
+   * pane should begin fading out (in milliseconds).
+   */
+  private hideDelay: number = 100;
+
+  /**
+   * The delay between when a touch event begins on the host element and when the
+   * hover pane should begin fading in (in milliseconds).
+   */
+  private longPressDelay: number = 600;
```

Comment:
Nice that you've broken out all these timings into well-documented properties! :)